src/basic/fs-util.c

```diff
@@ -1028,3 +1029,44 @@ int parse_cifs_service(
 
         return 0;
 }
+
+char *resolve_auto_dir(const char *path) {
+        _cleanup_free_ char *parent = NULL;
+        _cleanup_free_ char *glob_path = NULL;
+        _cleanup_globfree_ glob_t pglob = {};
+        int r;
+
+        if (!path)
+                return NULL;
+
+        log_debug("Trying to version %s.", path);
+        r = path_extract_directory(path, &parent);
```

Comment:
hmm, what I had in mind was that we could do this:

```
systemd-nspawn --image=/srv/myimages.auto.d/ -b
```

And that we'd then pick the newest image file inside that dir. Or:

```
systemd-nspawn --directory=/srv/myimages.auto.d -b
```

i.e. we'd specify the dir to look into. But the code above takes the *parent* of the specified dir, and that's what I don't grok?

Reply:
That would work too. So if the path is S_IFDIR, all files would be checked, but otherwise only items which match the path, using the final path component as a prefix. With the final path component, it's possible to put all RootImages etc. into same folder.

I want to use `foo.auto.d/systemd-networkd` as the path (matching only for example `foo.auto.d/systemd-networkd-1.0`, `foo.auto.d/systemd-networkd-2.0` etc.) as that's much simpler with LVM, otherwise I'd have to create a new VG for each service and that would kind of defeat the LVM idea.

Reply:
I don't understand what LVM has to do with this?

Reply:
I'm using LVM (when possible) for the RootImages. I don't think it's possible to make subdirectories (nested VGs) in LVM (something like `/dev/services/systemd-networkd.auto.d/` with LVs `ver-{1,2}`, instead of `/dev/services.auto.d/` containing all LVs ($SERVICE-{1,2}) for all services), so I can't have one `.auto.d/` per service without constructing new VGs for each service (`/dev/systemd-networkd.auto.d/`) which I don't want. Or maybe it is by stacking LVs over LVs, I have to check.

Reply:
Also directory without filename prefix works now